### ISSUE TITLE
weird zlib linkage issue on linux build for 0.33.{25,26} releases

### ISSUE THREAD

**Issue by chenrui333** [cube-js/cube#6738]
👋 while trying to upgrade cubejs-cli to the latest build, I ran into the linux linkage failure on linux build.

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/5298752490/jobs/9591329587
relates to https://github.com/Homebrew/homebrew-core/pull/133704

**Comment by ovr** [comment]
Hello @chenrui333,

Wow, surprise to see that Homebrew can be used in Linux.

Internally there is a module called `@cubejs-backend/native` which wraps native code on rust into Node.js. I will take a look at what is going on and which module requires `libz`.

Thanks

**Comment by chenrui333** [comment]
> Wow, surprise to see that Homebrew can be used in Linux.

yeah, we used to have linuxbrew (which is homebrew on linux), it got merged in the past years. technically homebrew can be run on any machines, just need to have build support for them :)

**Comment by iMichka** [comment]
Here is the file that is linked against system libz and which should not:
libexec/lib/node_modules/cubejs-cli/node_modules/@cubejs-backend/native/native/index.node

**Comment by ovr** [comment]
`cubejs-cli` uses `@cubejs-backend/schema-compiler` for validation, but It started to use `@cubejs-backend/native`.

There are two ways:

1. Use schema-compiler from project (load it dynamically)
2. Add `libz` as a requirement for the homebrew formula. (Then we should add python 3.11)

@keydunov @paveltiunov ☝️

**Comment by iMichka** [comment]
> Add libz as a requirement for the homebrew formula

I tried that. This does not work, as the precompiled library will still link to the system version.

**Comment by iMichka** [comment]
Do you think it would be possible to revert to `@cubejs-backend/schema-compiler`?
Or at least make the native version "optional" ?

As a package manager, we can't ship prebuilt binaries. We need to build everything from source on our own CI.

**Comment by paveltiunov** [comment]
@iMichka Does it fail during installation?

**Comment by iMichka** [comment]
No, build and installation is fine.

The issue you might get is:
- Runtime issues if /lib/x86_64-linux-gnu/libz.so.1 does not exist (maybe there are some docker images without lib, or maybe on some systems the path might differ. How about ARM or 32 bit paths?)
- If the build server `@cubejs-backend/native` is compromised, shipping an uncontrolled binary someone has build instead of Homebrew is not acceptable from a security point of view. We only trust our own build servers (they might get compromised, but that would be our issue, not yours :))
